src/lib.rs

```diff
@@ -671,135 +625,117 @@ impl EGraph {
+                                s.push_str(&name);
+                                for child in children
+                                    .iter()
+                                    .map(|child| termdag.to_string(termdag.get(*child)))
+                                {
+                                    s.push(',');
+                                    if child.contains(',')
+                                        || child.contains('\n')
+                                        || child.contains('"')
+                                    {
+                                        // escape commas and newlines in CSV
+                                        s.push('"');
+                                        s.push_str(&child.replace('"', "\"\""));
+                                        s.push('"');
+                                    } else {
+                                        s.push_str(&child);
+                                    }
+                                }
+
+                                if !out_is_unit {
+                                    s.push(',');
+                                    s.push_str(&termdag.to_string(&output));
+                                }
                             }
-                        }
-
-                        if !out_is_unit {
-                            s.push(',');
-                            s.push_str(&termdag.to_string(output));
+                            _ => panic!("Expect function_to_dag to return a list of apps."),
                         }
                     }
-                    _ => panic!("Expect function_to_dag to return a list of apps."),
-                }
-            } else {
-                let tuple_str = format!(
-                    "   {}{}",
-                    termdag.to_string(term),
-                    if !out_is_unit {
-                        format!(" -> {}", termdag.to_string(output))
-                    } else {
-                        "".into()
-                    },
-                );
-                log::info!("{}", tuple_str);
-                s.push_str(&tuple_str);
-            }
-            s.push('\n');
-        }
+                    s.push('\n');
+                    buf
+                } else {
+                    CommandOutput::PrintFunction(f.clone(), termdag, terms_and_outputs).to_string()
+                };
 
-        if !is_csv {
-            s.push_str(")\n");
-        }
-        match file {
-            Some(mut file) => {
-                file.write_all(buf.as_bytes())
-                    .expect("Failed to write to file");
+                file.write_all(lines.as_bytes())
+                    .expect("Error writing to file");
+                Ok(None)
             }
             None => {
-                self.print_msg(buf);
+                if is_csv {
+                    panic!("Cannot print a function to CSV if a filename is not provided")
```

Comment:
Why not? The user might be redirecting stdout and expects output to be in the CSV format

Reply:
Sure, I can change this to be allowed.